deisctl/backend/fleet/list_machines.go

```diff
@@ -0,0 +1,76 @@
+package fleet
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/coreos/fleet/machine"
+)
+
+const (
+	defaultListMachinesFields = "machine,ip,metadata"
+)
+
+var (
+	listMachinesFields = map[string]machineToField{
+		"machine": func(ms *machine.MachineState, full bool) string {
+			return machineIDLegend(*ms, full)
+		},
+		"ip": func(ms *machine.MachineState, full bool) string {
+			if len(ms.PublicIP) == 0 {
+				return "-"
+			}
+			return ms.PublicIP
+		},
+		"metadata": func(ms *machine.MachineState, full bool) string {
+			if len(ms.Metadata) == 0 {
+				return "-"
+			}
+			return formatMetadata(ms.Metadata)
+		},
+	}
+)
```

Comment:
I left it open so we could support other args for full machine ids or limiting columns.  I am not sure how much value we are adding by making a clone of the `fleetctl list-machines` command.